src/newgrf.cpp

```diff
@@ -6589,6 +6589,31 @@ static void SkipIf(ByteReader *buf)
 				result = rt != INVALID_ROADTYPE && RoadTypeIsTram(rt);
 				break;
 			}
+			case 0x13: {
+				uint r1 = GB(cond_val, 0, 8);
+				uint r2 = GB(cond_val, 8, 8);
+				if (r1 >= _cur.grffile->railtype_list.size() || r2 >= _cur.grffile->railtype_list.size()) {
+					result = false;
+					break;
+				}
+				RailType rt1 = GetRailTypeByLabel(_cur.grffile->railtype_list[r1]);
+				RailType rt2 = GetRailTypeByLabel(_cur.grffile->railtype_list[r2]);
+				result = rt1 != INVALID_RAILTYPE && rt2 != INVALID_RAILTYPE && rt1 == rt2;
+				break;
+			}
+			case 0x14: {
+				uint r1 = GB(cond_val, 0, 8);
+				uint r2 = GB(cond_val, 8, 8);
+				if (r1 >= _cur.grffile->railtype_list.size() || r2 >= _cur.grffile->railtype_list.size()) {
+					result = false;
+					break;
+				}
+				RailType rt1 = GetRailTypeByLabel(_cur.grffile->railtype_list[r1]);
+				RailType rt2 = GetRailTypeByLabel(_cur.grffile->railtype_list[r2]);
+				result = rt1 != INVALID_RAILTYPE && rt2 != INVALID_RAILTYPE && rt1 != rt2;
```

Comment:
This is not the negation of condition 0x13, which makes it weird in corner cases, and hard to implement in NML.

Reply:
that is intentional. in case one of the railtypes is undefined, both conditions return "false"

Reply:
maybe i'm too deep in formal logic here, but comparing <undefined> to <undefined> should never return true

Reply:
... and since we can't return `<undefined>`, the only option left is to return false